receiver/scraperhelper/scrapercontroller.go

```diff
@@ -34,7 +34,8 @@ import (
 // configuration. Scraper controller receivers can embed this struct, instead
 // of receiver.Settings, and extend it with more fields if needed.
 type ScraperControllerSettings struct {
-	CollectionInterval time.Duration `mapstructure:"collection_interval"`
+	CollectionInterval time.Duration  `mapstructure:"collection_interval"`
+	InitialDelay       *time.Duration `mapstructure:"initial_delay"`
```

Comment:
We should add a comment explaining why we use a pointer here (to distinguish between the zero value and the field being unset?). We don't generally use pointer fields, so I think we need to have a good reason to add it here

Reply:
That is a good point, I would if it is worth a type addition to confmap?

Reply:
I guess what would be better to ask is, what would make more sense:

- Define the method for `confmap.Marshaller` and see if it is being explicitly set like the mdatagen package does?
- Continue using the pointer method and just let it be set by `mapstructure`
- Using a value that isn't the zero value like `-1` that would imply isn't set and should default to X behaviour
- Include a new type in confmap that is safe to consume the zero value, doesn't require a pointer and is alligned to the sql package?

Reply:
> Define the method for confmap.Marshaller and see if it is being explicitly set like the mdatagen package does?

This sounds like the best option to me

Reply:
I want to understand the motivation in the first place? Are we not ok to change the default behavior to be an immediate scrape by default?

Reply:
Yeah, I was thinking from a user's perspective that behaviour suddenly changes and mislead to believe that the collector is now suddenly using more resources due to the potential overlap of collector getting ready and starting.

However, making this an opt out does make this a lot easier.